### PR TITLE
Controllerの移植：Eccube\Controller直下

### PR DESCRIPTION
**Pull request by hoand-vn**
## 概要(Overview・Refs Issue)
- Fixing all Unittest for symfony version

## 方針(Policy)
+ このPullRequestを作るにあたって考慮したものや除外し内容
  - When fixing Unittest, change some src
  - Using Dependency Injection
  - Almost files in Test/Web/* 

## 実装に関する補足(Appendix)
+ CartVvalidationTest.php still not finish 
+ ForgotControllerTest.php ( function testIndexWithPostAndVerify )  still not finish

## テスト（Test)
+ local

### CODE REVIEW

File: tests/Eccube/Tests/Web/AbstractShoppingControllerTestCase.php [hunk 1]
```diff
@@ -80,16 +80,17 @@ protected function scenarioInput($client, $formData)
     {
         $crawler = $client->request(
             'POST',
-            $this->app->path('shopping_nonmember'),
+            $this->generateUrl('shopping_nonmember'),
             array('nonmember' => $formData)
         );
-        $this->app['eccube.service.cart']->lock();
+        $this->container->get(CartService::class)->lock();
+//        $this->app['eccube.service.cart']->lock();
```
Comment (ryo-endo) [inline tests/Eccube/Tests/Web/AbstractShoppingControllerTestCase.php]: Please, delete.

File: tests/Eccube/Tests/Web/EntryControllerTest.php [hunk 2]
```diff
@@ -171,86 +164,86 @@ public function testConfirmWithModeNotFound()
...
 
     public function testActivateWithNotFound()
     {
         // debugはONの時に404ページ表示しない例外になります。
-        if($this->app['debug'] == true){
-            $this->setExpectedException('\Symfony\Component\HttpKernel\Exception\NotFoundHttpException');
-        }
-        $client = $this->createClient();
-        $crawler = $client->request('GET', $this->app['url_generator']->generate('entry_activate', array('secret_key' => 'aaaaa')));
+//        if($this->container->getParameter('kernel.debug') == true){
```
Comment (kiy0taka) [inline tests/Eccube/Tests/Web/EntryControllerTest.php]: Remove comment lines.

File: tests/Eccube/Tests/Web/ForgotControllerTest.php [hunk 3]
```diff
@@ -2,91 +2,103 @@
...
         $this->actual = $crawler->filter('div.ec-pageHeader > h1')->text();
         $this->verify();
 
-        $this->assertTrue($client->getResponse()->isSuccessful());
+        $this->assertTrue($this->client->getResponse()->isSuccessful());
     }
 
     public function testIndexWithPostAndVerify()
     {
+        $this->markTestIncomplete("reset_key can not found");
```
Comment (kiy0taka) [inline tests/Eccube/Tests/Web/ForgotControllerTest.php]: Call `$this-client->disableReboot()` at `setUp()`

File: tests/Eccube/Tests/Web/ForgotControllerTest.php [hunk 4]
```diff
@@ -97,34 +109,41 @@ public function testComplete()
...
-        // debugはONの時に403ページ表示しない例外になります。
-        if($this->app['debug'] == true){
-            $this->setExpectedException('\Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException');
-        }
-        $client = $this->createClient();
-            $crawler = $client->request(
-                'GET',
-                '/forgot/reset/a___aaa'
-            );
+        // Todo: Fail assert if debug = true
```
Comment (kiy0taka) [inline tests/Eccube/Tests/Web/ForgotControllerTest.php]: Remove comment lines.

File: src/Eccube/Controller/Admin/Product/ProductController.php [hunk 5]
```diff
@@ -148,12 +149,55 @@ class ProductController extends AbstractController
...
+        ProductClassRepository $productClassRepository,
+        ProductImageRepository $productImageRepository,
+        EntityManager $entityManager,
+        TaxRuleRepository $taxRuleRepository,
+        CategoryRepository $categoryRepository,
+        ProductRepository $productRepository,
+        BaseInfo $BaseInfo,
+        PageMaxRepository $pageMaxRepository,
+        ProductStatusRepository $productStatusRepository,
+        FormFactory $formFactory,
```
Comment (kiy0taka) [inline src/Eccube/Controller/Admin/Product/ProductController.php]: Use AbstractController's property.

File: src/Eccube/Controller/Admin/Product/ProductController.php [hunk 6]
```diff
@@ -148,12 +149,55 @@ class ProductController extends AbstractController
...
+        Session $session,
```
Comment (kiy0taka) [inline src/Eccube/Controller/Admin/Product/ProductController.php]: Use AbstractController's property.

File: src/Eccube/Controller/CartController.php [hunk 7]
```diff
@@ -45,41 +44,57 @@
...
 
+    /**
+     * CartController constructor.
+     * @param EventDispatcherInterface $eventDispatcher
+     * @param ProductClassRepository $productClassRepository
+     * @param PurchaseFlow $purchaseFlow
+     * @param CartService $cartService
+     */
+    public function __construct(
+        EventDispatcherInterface $eventDispatcher,
```
Comment (kiy0taka) [inline src/Eccube/Controller/CartController.php]: Use AbstractController's property.

File: src/Eccube/Controller/EntryController.php [hunk 8]
```diff
@@ -24,100 +24,128 @@
...
+        CustomerStatusRepository $customerStatusRepository,
+        MailService $mailService,
+        BaseInfoRepository $BaseInfo,
+        EntityManagerInterface $entityManager,
+        EventDispatcherInterface $eventDispatcher,
+        FormFactoryInterface $formFactory,
+        CustomerRepository $customerRepository,
+        EncoderFactoryInterface $encoderFactory,
+        ValidatorInterface $validatorInterface,
+        TokenStorageInterface $tokenStorage
```
Comment (kiy0taka) [inline src/Eccube/Controller/EntryController.php]: Use AbstractController's property.

**Comment by kiy0taka** [comment]
Now I merged 2 PRs.

+ #2769 Dependency Injection for `$BaseInfo`.
+ #2768 Use Mock CsrfTokenManager for Unit tests.

Could you apply these changes to this PR?

**Comment by kiy0taka** [comment]
Now AbstractController has `$eccubeConfig` / `$entityManager`.
https://github.com/EC-CUBE/ec-cube/pull/2770

**Comment by hoand-vn** [comment]
fixed some comments,
other comments will be fixed base on #2769 #2768 #2770  and make other PR.

**Comment by kiy0taka** [comment]
@hoand-vn 👍

**Pull request merged**